### PR TITLE
Health Implementation F5Status Merge

### PR DESCRIPTION
**Pull request by njuettner**

### CODE REVIEW

File: client/health.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	case "gtm":
+		trafficManager = GTMServer
+	default:
+		respondWithStatus(http.StatusNotFound, tm, nil, "TrafficManager not found", conf.Documentation["healthdocumentationuri"], c)
+		return
+	}
+	wideIPChan := make(chan gtm.HealthWideIPChannel)
+	firstMatchTrafficManager = snmp.CheckConnection(trafficManager, firstMatchTrafficManager)
+	go snmp.GetGTMWideIP(firstMatchTrafficManager, wideIPChan)
+	wch := <-wideIPChan
```
Comment (Raffo) [inline client/health.go]: Does it really make sense to use a goroutine if you then immediately wait for the response on the channel? How is this supposed to be better than just a function/method call?

File: client/health.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	for i, _ := range wch.Name {
+		status := util.CheckWideIPStatus(wch.Status[i].Value.(string))
```
Comment (Raffo) [inline client/health.go]: Do we really need this type assertions? They can easily bring to errors.

Comment (njuettner) [inline client/health.go]: Unfortunately yes, I looked in several Go SNMP packages and they doing almost the same. 

The response value is always given as an interface{} depending on the PDU response from the SNMP server.

File: client/health.go (hunk 1 above)
Comment (njuettner) [inline client/health.go]: Right, my plan was to use go channel to ask every GTM/ITM whats the current status for wideips and pools, however only one is asked to give status of wideips and pools for now.

GSLB servers is correct.

File: client/health.go (hunk 2 above)
Comment (Raffo) [inline client/health.go]: Ok good, so the only way to make it reliable is to have a pretty good testing.

File: client/health.go (hunk 1 above)
Comment (Raffo) [inline client/health.go]: :+1:

File: client/health.go (hunk 2 above)
Comment (szuecs) [inline client/health.go]: Maybe do a nil check on wch.Status[i] , before calling Value and type cast?

**Comment by LappleApple** [comment]
@Raffo @njuettner @szuecs Any reason why this PR hasn't been merged?

**Comment by Raffo** [comment]
Not that I know. I didn't review PRs on this repos anymore due to change of teams. Maybe @njuettner can clarify the status.

**Pull request merged**